include/xtensor/xmath.hpp

```diff
@@ -1891,7 +1891,7 @@ XTENSOR_INT_SPECIALIZATION_IMPL(FUNC_NAME, RETURN_VAL, unsigned long long);
         }
 
         using result_type = promote_type_t<double, typename std::decay_t<E>::value_type>;
-        result_type scl = static_cast<result_type>(sum(weights, xt::evaluation_strategy::immediate{})());
+        result_type scl = static_cast<result_type>(sum(weights, ax, xt::evaluation_strategy::immediate{})());
```

Comment:
you're totally right! removing the `result_type` and setting it to auto + removing the (empty) bracket at the end (which returns the scalar at `0`) fixes the issue.

`auto scl = sum(weights, ax, xt::evaluation_strategy::immediate{});`

thanks for spotting this + sorry for the oversight.